### PR TITLE
K8SPS-113: specify memory size as a template parameter in the config

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPS-113](https://jira.percona.com/browse/K8SPS-113) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPS-113

I decided to use a [pongo2](https://github.com/flosch/pongo2) template engine. Currently, only the `contaninerMemoryLimit` variable is available for use in templates. So, to use for example `contaninerMemoryLimit*3/4`, we should write `{{containerMemoryLimit * 3/4}}`. It's not possible to use templates if memory is not specified in `mysql.resources`.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/init-deploy.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/sidecars.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/users.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/limits.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/0ebaf3a/gr-init-deploy.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/0ebaf3abb4b2b6a2eeb9c2afd012ed9929f56fbe
image: `perconalab/percona-server-mysql-operator:PR-120-0ebaf3a`

File: deploy/cr.yaml [hunk 1]
```diff
@@ -48,6 +48,7 @@ spec:
 
 #    configuration: |
 #      max_connections=250
+#      innodb_buffer_pool_size={{int64 (mulf .Memory (divf 3 4))}}
```
Comment (hors) [inline deploy/cr.yaml:51]: @spron-in what do you think about it?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/users.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/34ad8b3/config.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/34ad8b380207021e2bb5a6195339a131746684e7
image: `perconalab/percona-server-mysql-operator:PR-120-34ad8b3`

**Review by hors — CHANGES_REQUESTED** [review]
@pooknull please add test for this feature(you can use auto-config test).

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/ca179a5d2daaa403d479de637597ce8c2a6cab73
image: `perconalab/percona-server-mysql-operator:PR-120-ca179a5`

**Comment by pooknull** [comment]
> @pooknull please add test for this feature(you can use auto-config test).

https://github.com/percona/percona-server-mysql-operator/pull/120/commits/cab26510743b62d7776a45d859f4865c3524ae6d

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/monitoring.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/scaling.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/service-per-pod.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/users.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/limits.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cab2651/config.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/cab26510743b62d7776a45d859f4865c3524ae6d
image: `perconalab/percona-server-mysql-operator:PR-120-cab2651`

**Review by hors — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/monitoring.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/config.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/limits.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/df222e0/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/df222e0c9c52b153a37cc7f6e004087a794554ac
image: `perconalab/percona-server-mysql-operator:PR-120-df222e0`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/semi-sync.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/config.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/scaling.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/service-per-pod.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/users.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/54e8728/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/54e872891172e633222543ac8c510651f888b53c
image: `perconalab/percona-server-mysql-operator:PR-120-54e8728`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/config.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/limits.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/cfb3155/users.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/cfb3155d2ae2232f8f64a1382b60a26a9e1850fb
image: `perconalab/percona-server-mysql-operator:PR-120-cfb3155`

**Review by inelpandzic — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/init-deploy.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/auto-config.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/scaling.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/users.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/11a49b2/tls-cert-manager.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/11a49b2220d8e7b85ba3a29f4509d4df3a1aa1c2
image: `perconalab/percona-server-mysql-operator:PR-120-11a49b2`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/semi-sync.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/config.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/users.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/ead3703/tls-cert-manager.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/ead3703bce1a10d22933cd435d92c68472ede5cc
image: `perconalab/percona-server-mysql-operator:PR-120-ead3703`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/limits.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/version-service.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-120/9140e87/users.log) |
| We run 14 out of 14|

commit: https://github.com/percona/percona-server-mysql-operator/pull/120/commits/9140e8760c3c1b2d14e9e5bdae036fabf4fde5b9
image: `perconalab/percona-server-mysql-operator:PR-120-9140e87`

**Pull request merged**